pyglet/math.py

```diff
@@ -640,49 +687,56 @@ def __new__(cls, values=None) -> 'Mat3':
             `values` : tuple of float or int
                 A tuple or list containing 9 floats or ints.
         """
-        assert values is None or len(values) == 9, "A 3x3 Matrix requires 9 values"
-        return super().__new__(Mat3, values or (1.0, 0.0, 0.0,
-                                                0.0, 1.0, 0.0,
-                                                0.0, 0.0, 1.0))
+        new = super().__new__(Mat3, values)
+        assert len(new) == 9, "A 3x3 Matrix requires 9 values"
+        return new
 
-    def scale(self, sx: float, sy: float):
-        return self @ (1.0 / sx, 0.0, 0.0, 0.0, 1.0 / sy, 0.0, 0.0, 0.0, 1.0)
+    def scale(self, sx: float, sy: float) -> Mat3:
+        return self @ Mat3((1.0 / sx, 0.0, 0.0, 0.0, 1.0 / sy, 0.0, 0.0, 0.0, 1.0))
 
-    def translate(self, tx: float, ty: float):
-        return self @ (1.0, 0.0, 0.0, 0.0, 1.0, 0.0, -tx, ty, 1.0)
+    def translate(self, tx: float, ty: float) -> Mat3:
+        return self @ Mat3((1.0, 0.0, 0.0, 0.0, 1.0, 0.0, -tx, ty, 1.0))
 
-    def rotate(self, phi: float):
+    def rotate(self, phi: float) -> Mat3:
         s = _math.sin(_math.radians(phi))
         c = _math.cos(_math.radians(phi))
-        return self @ (c, s, 0.0, -s, c, 0.0, 0.0, 0.0, 1.0)
+        return self @ Mat3((c, s, 0.0, -s, c, 0.0, 0.0, 0.0, 1.0))
 
-    def shear(self, sx: float, sy: float):
-        return self @ (1.0, sy, 0.0, sx, 1.0, 0.0, 0.0, 0.0, 1.0)
+    def shear(self, sx: float, sy: float) -> Mat3:
+        return self @ Mat3((1.0, sy, 0.0, sx, 1.0, 0.0, 0.0, 0.0, 1.0))
 
-    def __add__(self, other) -> 'Mat3':
-        assert len(other) == 9, "Can only add to other Mat3 types"
-        return Mat3(tuple(s + o for s, o in zip(self, other)))
+    def __add__(self, other: Mat3) -> Mat3:
+        if not isinstance(other, Mat3):
+            raise TypeError("Can only add to other Mat3 types")
+        return Mat3(s + o for s, o in zip(self, other))
 
-    def __sub__(self, other) -> 'Mat3':
-        assert len(other) == 9, "Can only subtract from other Mat3 types"
-        return Mat3(tuple(s - o for s, o in zip(self, other)))
+    def __sub__(self, other: Mat3) -> Mat3:
+        if not isinstance(other, Mat3):
+            raise TypeError("Can only add to other Mat3 types")
```

Comment:
Looks like it should be "subtract" here instead of "add".

Reply:
Fixed.